src/libraries/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs

```diff
@@ -17,8 +18,18 @@ namespace System.Xml.Serialization
 
     internal class ReflectionXmlSerializationReader : XmlSerializationReader
     {
-        private static TypeDesc StringTypeDesc { get; set; } = (new TypeScope()).GetTypeDesc(typeof(string));
-        private static TypeDesc QnameTypeDesc { get; set; } = (new TypeScope()).GetTypeDesc(typeof(XmlQualifiedName));
+        private static readonly TypeDesc StringTypeDesc;
+        private static readonly TypeDesc QnameTypeDesc;
+
+        [RequiresUnreferencedCode("Calls GetTypeDesc")]
```

Comment:
What does `RequiresUnreferencedCode` on a static ctor mean / do?

A different approach to suppress the warning would be to use the `Target` property, like in the following:

https://github.com/dotnet/runtime/blob/2d0fe49001423b18b520c9fae072c81e7a11c50c/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs#L18-L20

Reply:
It doesn't look like usage of the type with RUC on the static constructor warns today, but it seems like it should. Personally, I would expect it to work similarly to putting `ObsoleteAttribute` on a type -- every static usage of the type (including constructor) would warn, but instance invocations wouldn't.

Reply:
> It doesn't look like usage of the type with RUC on the static constructor warns today, but it seems like it should.

I think Mateo was working on it recently. I have a separate conversation on whether my understanding of this will match the implementation, but I think for classes, marking the cctor as RUC should generate warnings on all constructors and static methods on the type that then need to be suppressed or propagated. The drop of illink in this repo probably doesn't have Mateo's changes yet or the annotation doesn't do what I think it needs to do (will find out soon).

Reply:
Even though it might work, the experience will be weird - it's basically equivalent to putting the RUC on the type (if we supported that). Is it really the case that the type is impossible to use safely with trimming - even for a little bit?

Reply:
> Is it really the case that the type is impossible to use safely with trimming - even for a little bit?

In this specific case, I believe we can safely suppress these warning since the hard-coded types used here `string` and `XmlQualifiedName` should be safe to pass to `GetTypeDesc`. Looking at the annotation for that method, it looks like it only gets into unsafe code if it calls `ImportTypeDesc`, which it should never do for these two types because they are considered "primitive" types:

https://github.com/dotnet/runtime/blob/f64246ce08fb7a58221b2b7c8e68f69c02522b0d/src/libraries/System.Private.Xml/src/System/Xml/Serialization/Types.cs#L493-L504